lxd/instance/drivers/driver_qemu.go

```diff
@@ -5219,45 +5219,32 @@ func (d *qemu) renderState(statusCode api.StatusCode) (*api.InstanceState, error
 	pid, _ := d.pid()
 
 	if d.isRunningStatusCode(statusCode) {
-		// Try and get state info from agent.
-		status, err = d.agentGetState()
-		if err != nil {
-			if err != errQemuAgentOffline {
-				d.logger.Warn("Could not get VM state from agent", log.Ctx{"err": err})
-			}
+		val := d.expandedConfig["security.agent.metrics"]
 
-			// Fallback data if agent is not reachable.
-			status = &api.InstanceState{}
+		if !shared.IsTrue(val) {
```

Comment:
`if !shared.IsTrue(d.expandedConfig["security.agent.metrics"]) {` would be fine here rather than having the extra generic `val` variable.

Reply:
Also, shouldn't this default to true if empty/not set?

Reply:
And can you re-arrange it so its `if true {} else {}` rather than `if !true {} else {}`.

Reply:
> Also, shouldn't this default to true if empty/not set?

It does. That's why I did `if !shared.True()`. If `val` is `true` or empty, it'll go to the `else` branch.

Reply:
If I set `security.agent.metrics` is not set or empty, `shared.IsTrue` will return false, causing it to not use the agent, but my understand is that if its unset/empty then the agent should be used.

The logic should be the same as `if val == "" || shared.IsTrue(val) {` is below, right (or wrapped in a function to avoid duplicating it)?